lib/sinon.js

```diff
@@ -7,7 +7,21 @@
  * Copyright (c) 2010-2013 Christian Johansen
  */
 "use strict";
-module.exports = require("./sinon/util/core");
+
+var match = require("./sinon/match");
+
+module.exports = exports = require("./sinon/util/core");
+
+exports.extend = require("./sinon/extend");
+exports.match = match;
+exports.typeOf = require("./sinon/typeOf");
+
+/*
+ * allow deepEqual to check equality of matchers through
+ * dependency injectection. Otherwise we get a circular
+ * dependency
+ */
+exports.deepEqual.match = match;
```

Comment:
@fearphage Discussion by text is hard - not sure I am reading you correctly :-) So to clear things up; did you somehow create that same snippet of code in some earlier commit that was then purged in some rebase, or was something like this present in the existing code. In both cases, I clearly missed it somehow, and in any case, it shows that we are living in the same code culture. 

I also found `withMatcher` to be suboptimal. Not sure if my rename to `.using` is any better, but I think it reads pretty well (?). 

If it seems fine by you, might as well hear if the API fits the mind of @mroderick